libraries/AP_Scripting/drivers/mount-djirs2-driver.lua

```diff
@@ -0,0 +1,734 @@
+local CAN_P2_BITRATE = Parameter("CAN_P2_BITRATE")      -- If using CAN2, should be 1000000
+local CAN_D2_PROTOCOL = Parameter("CAN_D2_PROTOCOL")    -- If using CAN2, should be 10:Scripting
+local MNT1_TYPE = Parameter("MNT1_TYPE")                -- should be 9:Scripting
+local DJIR_DEBUG = Parameter("DJIR_DEBUG")              -- debug level. 0:disabled 1:enabled 2:enabled with attitude reporting
+
+-- message definitions
+local HEADER = 0xAA
+local RETURN_CODE = {SUCCESS=0x00, PARSE_ERROR=0x01, EXECUTION_FAILED=0x02, UNDEFINED=0xFF}
+
+-- parsing state definitions
+local PARSE_STATE_WAITING_FOR_HEADER        = 0
+local PARSE_STATE_WAITING_FOR_VERLENGTH     = 1
+local PARSE_STATE_WAITING_FOR_DATA          = 2
+
+-- other parsing definitions
+local CAN_PACKET_LENGTH_MAX = 8         -- CAN packet maximum length
+local SERIAL_PACKET_LENGTH_MAX = 32     -- serial packet maximum length.  used to sanity check length of incoming messages
+local SERIAL_PACKET_LENGTH_MIN = 16     -- serial packet minimum length.  used to sanity check sends
+
+-- local variables and definitions
+local driver                            -- CAN bus
+local initialised = false               -- true once connection to gimbal has been initialised
+local parse_state = PARSE_STATE_WAITING_FOR_HEADER  -- parse state
+local parse_length = 0                  -- incoming message's packet length
+local parse_buff = {}                   -- message buffer holding roll, pitch and yaw angles from gimbal
+local parse_bytes_recv = 0              -- message buffer length.  count of the number of bytes received in the message so far
+local last_send_seq = 0                 -- last sequence number sent
+local last_req_attitude_ms = 0          -- system time of last request for attitude
+local last_set_attitude_ms = 0          -- system time of last set attitude call
+local REPLY_TYPE = {NONE=0, ATTITUDE=1, POSITION_CONTROL=2, SPEED_CONTROL=3} -- enum of expected reply types
+local expected_reply = REPLY_TYPE.NONE  -- currently expected reply type
+local expected_reply_ms = 0             -- system time that reply is first expected.  used for timeouts
+
+-- parsing status reporting variables
+local last_print_ms = 0                 -- system time that debug output was last printed
+local bytes_read = 0                    -- number of bytes read from gimbal
+local bytes_written = 0                 -- number of bytes written to gimbal
+local bytes_error = 0                   -- number of bytes read that could not be parsed
+local msg_ignored = 0                   -- number of ignored messages (because frame id does not match)
+local write_fails = 0                   -- number of times write failed
+local execute_fails = 0                 -- number of times that gimbal was unable to execute the command
+local reply_timeouts = 0                -- number of timeouts waiting for replies
+
+local crc16_lookup = {
```

Comment:
where did his 16bit crc table come from?

Reply:
![image](https://user-images.githubusercontent.com/4782875/215630002-f49b827d-0c73-4001-a5ab-0ff3e919235c.png)

Turns out this uses our crc_crc16_ibm() with a different seed. Take a look at this:
https://www.lddgo.net/en/encrypt/crc

Reply:
This one is this according to this tool here,
http://www.sunshine2k.de/coding/javascript/crc/crc_js.html

![image](https://user-images.githubusercontent.com/69225461/215630328-1edbce65-e90d-41d6-b8a6-b5a3e34ebfed.png)

Reply:
Thanks for this.  I will replace the crc methods with bindings to AP's existing crc functions in a follow-up PR.